### PR TITLE
Integration and unit tests

### PR DESCRIPTION
**Pull request by bmalecki**
Fix integration and unit test. Some code and config refactor.

### CODE REVIEW

File: it-test/src/test/java/io/knotx/forms/test/integration/FormsIntegrationTest.java [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+  @KnotxApplyConfiguration("formsStack.conf")
+  public void callForms_validKnotContextResult(
+      VertxTestContext vertxTestContext, Vertx vertx)
+      throws IOException, URISyntaxException {
+
+    callWithAssertions(vertxTestContext, vertx, "fragment_form_self_in.txt",
+        knotContext -> {
+          JsonObject context = knotContext.getFragments().iterator().next().context();
+
+          Assertions.assertTrue(context.containsKey("action"));
```
Comment (tomaszmichalak) [inline it-test/src/test/java/io/knotx/forms/test/integration/FormsIntegrationTest.java]:
```suggestion
          Assertions.assertTrue(context.containsKey("forms"));
```

File: it-test/src/test/java/io/knotx/forms/test/integration/FormsIntegrationTest.java [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+          Assertions.assertTrue(context.getJsonObject("action").containsKey("_result"));
```
Comment (tomaszmichalak) [inline it-test/src/test/java/io/knotx/forms/test/integration/FormsIntegrationTest.java]:
```suggestion
          Assertions.assertTrue(context.getJsonObject("forms").containsKey("_result"));
```

File: it-test/src/test/java/io/knotx/forms/test/integration/FormsIntegrationTest.java [hunk 3]
```diff
@@ -0,0 +1,103 @@
...
+          Assertions.assertTrue(
+              context.getJsonObject("action").getJsonObject("_result").containsKey("mock"));
```
Comment (tomaszmichalak) [inline it-test/src/test/java/io/knotx/forms/test/integration/FormsIntegrationTest.java]:
```suggestion
              context.getJsonObject("forms").getJsonObject("_result").containsKey("mock"));
```

**Review by tomaszmichalak — CHANGES_REQUESTED** [review]

File: it-test/src/test/java/io/knotx/forms/test/integration/FormsIntegrationTest.java [hunk 4]
```diff
@@ -0,0 +1,103 @@
...
+
+  @Test
+  @KnotxApplyConfiguration("formsStack.conf")
+  public void callForms_validKnotContextResult(
+      VertxTestContext vertxTestContext, Vertx vertx)
+      throws IOException, URISyntaxException {
+
+    callWithAssertions(vertxTestContext, vertx, "fragment_form_self_in.txt",
+        knotContext -> {
+          JsonObject context = knotContext.getFragments().iterator().next().context();
```
Comment (tomaszmichalak) [inline it-test/src/test/java/io/knotx/forms/test/integration/FormsIntegrationTest.java]: Please add HTTP method to the request. We should validate scenarios:
1. GET request that renders form details without Knot.x specific tags.
2. POST request that calls the mock adapter and passes the flow to the next Knot.
3. POST request that calls the mock adapter and based on signal redirects to the different page.

File: it-test/src/test/java/io/knotx/forms/test/integration/FormsIntegrationTest.java [hunk 5]
```diff
@@ -0,0 +1,103 @@
...
+
+@Ignore
+@ExtendWith(KnotxExtension.class)
+public class FormsIntegrationTest {
+
+  private final static String CORE_MODULE_EB_ADDRESS = "knotx.knot.forms";
+  private final static String FORM_NAME = "someId456";
+
+  @Test
+  @KnotxApplyConfiguration("formsStack.conf")
```
Comment (tomaszmichalak) [inline it-test/src/test/java/io/knotx/forms/test/integration/FormsIntegrationTest.java]: I think we can define the example configuration in `conf` folder:
```
conf/
       formStack.conf
       includes/
                    formsKnot.conf
                    exampleFormAdatper.conf 
```
and then in the test method specify `formsStack.conf` that contains default settings and add `testScenarioXXX.conf` that overrides defaults. The example can be found here: io.knotx.example.monolith.SampleApplicationIntegrationTest#whenRequestingLocalSimplePageWithGetCustomSymbol_expectLocalSimpleHtml.

Additionally we can specify ExampleFormAdapter behaviour based on expected scenarios. So the configuration can be:

exampleFormAdapter.address = ...
exampleFormAdapter.testStrategy = GET_REQUEST

And in the Example Form Adapter we can implement all cases. Please not that both Example Form Adapter and `testScenarioXXX.conf` are defined in tests scope.

File: conf/formsStack.conf [hunk 6]
```diff
@@ -3,34 +3,30 @@ modules = [
   ########### Forms ###########
   "formsKnot=io.knotx.forms.core.FormsKnot"
 
-  ########### Custom ###########
-  ### Change to your adapter ###
-  "formsAdapter=com.example.forms.adapter.http.FormsAdapterVerticle"
+  ####### Forms Adpater #######
```
Comment (tomaszmichalak) [inline conf/formsStack.conf:6]: Please add links to those configuration files in `README.md`.

**Review by tomaszmichalak — CHANGES_REQUESTED** [review]

File: conf/formsStack.conf (hunk 6 above)
Comment (bmalecki) [reply conf/formsStack.conf:6]: Added

**Review by bmalecki — COMMENTED** [review]
Added test configuration resources

File: core/src/main/java/io/knotx/forms/core/FormsKnotProxy.java [hunk 7]
```diff
@@ -142,11 +142,11 @@ private KnotContext processAdapterResponse(KnotContext knotContext, List<FormEnt
   private KnotContext routeToNextKnotResponse(ClientResponse clientResponse,
       KnotContext knotContext, List<FormEntity> forms, FormEntity form) {
     LOGGER.debug("Request next transition to [{}]", DEFAULT_TRANSITION);
-    JsonObject actionContext = new JsonObject()
+    JsonObject formsContext = new JsonObject()
         .put("_result", new JsonObject(clientResponse.getBody().toString()))
         .put("_response", clientResponse.toMetadataJson());
 
-    form.fragment().context().put("action", actionContext);
+    form.fragment().context().put("forms", formsContext);
```
Comment (tomaszmichalak) [inline core/src/main/java/io/knotx/forms/core/FormsKnotProxy.java]: Extract "forms" to constant in API module.

File: core/src/main/java/io/knotx/forms/core/domain/FormConstants.java [hunk 8]
```diff
@@ -26,7 +26,7 @@
   static final String FORM_ADAPTER_ATTR = "data-knotx-forms-adapter";
```
Comment (tomaszmichalak) [inline core/src/main/java/io/knotx/forms/core/domain/FormConstants.java]: "data-knotx-forms-adapter" -> "data-knotx-forms-adapter-name"

File: core/src/main/java/io/knotx/forms/core/domain/FormConstants.java [hunk 9]
```diff
@@ -26,7 +26,7 @@
...
   static final String FORM_ADAPTER_PARAMS = "data-knotx-forms-adapter-params";
 
-  static final String ACTION_FORM_ATTRIBUTES_PATTERN = "data-knotx-.*";
+  static final String FORM_ATTRIBUTES_PATTERN = "data-knotx-.*";
```
Comment (tomaszmichalak) [inline core/src/main/java/io/knotx/forms/core/domain/FormConstants.java]: I think we can assume that we search for "data-knotx-forms-.*".

File: it-test/src/test/java/io/knotx/forms/test/integration/FormsIntegrationTest.java [hunk 10]
```diff
@@ -0,0 +1,130 @@
...
+      VertxTestContext vertxTestContext, Vertx vertx)
+      throws IOException, URISyntaxException {
+    KnotContext message = payloadMessage("fragment_form_self_in.txt");
+    message.getClientRequest().setMethod(HttpMethod.POST);
+
+    rxProcessWithAssertions(vertxTestContext, vertx, message, this::assertValidFormsContext);
+  }
+
+  @Test
+  @KnotxApplyConfiguration("formsStack_GET.conf")
```
Comment (tomaszmichalak) [inline it-test/src/test/java/io/knotx/forms/test/integration/FormsIntegrationTest.java]: `@KnotxApplyConfiguration({"formsStack.conf","adapterGET.conf"})`

?

**Review by tomaszmichalak — CHANGES_REQUESTED** [review]

File: api/src/main/java/io/knotx/forms/api/FormFragmentConstants.java [hunk 11]
```diff
@@ -0,0 +1,8 @@
+package io.knotx.forms.api;
+
+public class FormFragmentConstants {
```
Comment (tomaszmichalak) [inline api/src/main/java/io/knotx/forms/api/FormFragmentConstants.java]: final

Comment (tomaszmichalak) [inline api/src/main/java/io/knotx/forms/api/FormFragmentConstants.java]:
```suggestion
public final class FormFragmentConstants {
```

**Review by tomaszmichalak — APPROVED** [review]

**Pull request merged**